control-plane/msp-operator/src/main.rs

```diff
@@ -0,0 +1,910 @@
+            }
+            UrlPath::Pool(name) => {
+                url.set_path(&format!("v0/nodes/{}/pools/{}", self.spec.node, name))
+            }
+        };
+
+        Ok(url)
+    }
+
+    /// helper function to set the path for the URL we want to perform the GET
+    /// operation on
+    fn get(&self, n: UrlPath) -> Result<RequestBuilder, Error> {
+        let url = self.as_url(n)?;
+        Ok(self.ctx.http.get(url))
+    }
+
+    /// helper for setting the path of the URL we want to PUT operaiton on.
+    fn put(&self, n: UrlPath) -> Result<RequestBuilder, Error> {
+        // we only do puts for to one specific path
+        if matches!(n, UrlPath::Pool(_)) {
+            let url = self.as_url(n)?;
+            return Ok(self.ctx.http.put(url));
+        }
+
+        Err(Error::Noun {})
+    }
+
+    /// helper for setting the path of the URL we want to PUT operaiton on.
+    fn delete(&self, n: UrlPath) -> Result<RequestBuilder, Error> {
+        // we only do delete for to one specific path
+        if matches!(n, UrlPath::Pool(_)) {
+            let url = self.as_url(n)?;
+            return Ok(self.ctx.http.delete(url));
+        }
+
+        Err(Error::Noun {})
+    }
+
+    /// Patch the given MSP status to the state provided. When not online the
+    /// size should be asumed to be zero.
+    async fn patch_status(&self, status: MayastorPoolStatus) -> Result<MayastorPool, Error> {
+        let status = json!({ "status": status });
+
+        let ps = PatchParams::apply(WHO_AM_I);
+
+        let o = self
+            .api()
+            .patch_status(&self.name(), &ps, &Patch::Merge(&status))
+            .await
+            .map_err(|source| Error::Kube { source })?;
+
+        debug!(name = ?o.name(), old = ?self.status, new =?o.status, "status changed");
+
+        Ok(o)
+    }
+
+    /// Create a pool when there is no status found. When no status is found for
+    /// this resource it implies that it does not exist yet and so we create
+    /// it. We set the state of the of the object to Creating, such that we
+    /// can track the its progress
+    async fn start(&self) -> Result<ReconcilerAction, Error> {
+        let _ = self.patch_status(MayastorPoolStatus::default()).await?;
+        Ok(ReconcilerAction {
+            requeue_after: None,
+        })
+    }
+
+    /// Mark the resource as errorerd which is its final state. A pool in the
+    /// error state will not be deleted.
+    async fn mark_error(&self) -> Result<ReconcilerAction, Error> {
+        let _ = self.patch_status(MayastorPoolStatus::error()).await?;
+
+        error!(name = ?self.name(),"status set to error");
+        Ok(ReconcilerAction {
+            requeue_after: None,
+        })
+    }
+
+    /// patch the resource state to missing
+    async fn is_missing(&self) -> Result<ReconcilerAction, Error> {
+        self.patch_status(MayastorPoolStatus::default()).await?;
```

Comment:
We are using the default setting which will set it to "Creating" rather than "Missing".

Reply:
There is no missing state but good catch!. I've updated the comment. By setting the pool back into the creating state -- it means "we are creating the pool and driving it to... ". As we don't do anything here, we just wait for it to come back online.